src/samples/p2p/js/sc.websocket.js

```diff
@@ -71,14 +58,13 @@ function SignalingChannel() {
     wsServer.on('connect', function() {
       reconnectTimes = 0;
       console.info('Connected to websocket server.');
-    });
-
-    wsServer.on('server-authenticated', function(data) {
-      console.log('Authentication passed. User ID: ' + data.uid);
-      if (connectPromise) {
-        connectPromise.resolve(data.uid);
-      }
-      connectPromise = null;
+      wsServer.emit('authentication', {token: loginInfo.token}, (data) => {
+        console.log('Authentication passed. User ID: ' + data.uid);
+        if (connectPromise) {
```

Comment:
check status for resolving/reject connectPromise.

Reply:
`connectPromise` is `null` after it is settled.

Reply:
I would expect this:
wsServer.emit('authentication', {token: loginInfo.token}, (status, data) => {.... }

Reply:
Done. Parsing `data` before resolving `connectPromise`.